lib/scripts.js

```diff
@@ -371,61 +386,265 @@ function createList(fileName) {
+    if (!(await npmVersionExists(adapterName, version))) {
+        throw new Error(`Cannot add ${adapterName}@${version} to stable because it is not yet on npm!`);
+    }
+
+    const stable = await readStableRepo();
+    if (adapterName in stable) throw new Error(`${adapterName} is already in stable!`);
+    const latest = await readLatestRepo();
+    if (!(adapterName in latest)) throw new Error(`Cannot add ${adapterName} to stable because it is not yet in the latest repo!`);
+
+    stable[adapterName] = {
+        ...latest[adapterName],
+        version
+    };
+
+    await writeStableRepo(stable);
+}
+
+async function removeDates() {
+    const stable = await readStableRepo();
+    Object.keys(stable).forEach(name => {
+        delete stable[name].published;
+        delete stable[name].versionDate;
+    });
+    await writeStableRepo(stable);
+
+    const latest = await readStableRepo();
+    Object.keys(latest).forEach(name => {
+        delete latest[name].published;
+        delete latest[name].versionDate;
     });
-    fs.writeFileSync(__dirname + '/../sources-dist.json', JSON.stringify(sortRepo(sources), null, 2));
+    await writeLatestRepo(latest);
 }
 
+function fail(reason) {
+    console.error();
+    console.error('ERROR: ' + reason);
+    console.error();
+    process.exit(1);
+}
 
 if (module.exports && module.parent) {
-    module.exports.init = init;
-    module.exports.sort = sort;
-    module.exports.list = createList;
-    module.exports.nodates = removeDates;
+    module.exports = {
+        init,
+        sort,
+        list: createList,
+        nodates: removeDates,
+        addToLatest,
+        addToStable,
+    };
 } else {
+    const argv = require('minimist')(process.argv.slice(2));
     // update versions for all adapter, which do not have the version
-    if (process.argv.indexOf('--init') !== -1) {
-        init(() => process.exit());
-    } else if (process.argv.indexOf('--nodates') !== -1) {
+    if (argv._.includes('init')) {
+        init().then(() => process.exit());
+    } else if (argv._.includes('nodates')) {
         removeDates();
-    } else if (process.argv.indexOf('--sort') !== -1) {
+    } else if (argv._.includes('sort')) {
         sort();
-    } else if (process.argv.indexOf('--list') !== -1) {
-        const file = process.argv.indexOf('--list');
-        if (process.argv[file + 1]) {
-            createList(process.argv[file + 1]);
+    } else if (argv._.indexOf('list') !== -1) {
+        const file = argv._[argv._.indexOf('list') + 1];
+        if (file) {
+            createList(file);
         } else {
             createList(__dirname + '/../list.html');
         }
+    } else if (argv._.includes('addToStable')) {
+        const {name, version} = argv;
+        if (typeof name !== 'string') fail('Please specify the adapter name!');
+        if (typeof version !== 'string') fail('Please specify the version to add to stable!');

```

Comment:
If the version not provided, we can get the latest automatically. The function is here anyway

Reply:
True. Will update it later